src/main/java/com/terraformersmc/terrestria/feature/trees/templates/SmallLogTree.java

```diff
@@ -0,0 +1,123 @@
+package com.terraformersmc.terrestria.feature.trees.templates;
+
+import com.mojang.datafixers.Dynamic;
+import com.terraformersmc.terraform.block.BareSmallLogBlock;
+import com.terraformersmc.terraform.block.SmallLogBlock;
+import net.minecraft.block.BlockState;
+import net.minecraft.state.property.BooleanProperty;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.MutableIntBoundingBox;
+import net.minecraft.world.ModifiableTestableWorld;
+import net.minecraft.world.gen.feature.AbstractTreeFeature;
+import net.minecraft.world.gen.feature.DefaultFeatureConfig;
+
+import java.util.Random;
+import java.util.Set;
+import java.util.function.Function;
+
+public class SmallLogTree extends AbstractTreeFeature<DefaultFeatureConfig> {
+
+	private BlockState log;
+	private BlockState leaves;
+
+	public SmallLogTree(Function<Dynamic<?>, ? extends DefaultFeatureConfig> function_1, boolean boolean_1, BlockState log, BlockState leaves) {
+		super(function_1, boolean_1);
+
+		this.log = log;
+		this.leaves = leaves;
+	}
+
+	@Override
+	protected boolean generate(Set<BlockPos> set, ModifiableTestableWorld modifiableTestableWorld, Random random, BlockPos blockPos, MutableIntBoundingBox mutableIntBoundingBox) {
+		return false;
+	}
+
+	public BlockState getLog() {
+		return log;
+	}
+
+	public BlockState getLeaves() {
+		return leaves;
+	}
+
+	protected void setBlockStateAndUpdate(Set<BlockPos> blocks, ModifiableTestableWorld world, BlockPos.Mutable origin, BlockState state, Direction direction, MutableIntBoundingBox boundingBox) {
+		BlockPos.Mutable pos = new BlockPos.Mutable(origin.offset(direction.getOpposite()));
+		if (getOriginalState(world, pos) != null) {
+			//Fix the previous block
+			setBlockState(blocks, world, pos, getOriginalState(world, pos).with(getStateFromDirection(direction), true), boundingBox);
+		}
+		pos.setOffset(direction);
+		//Place a new block and connect it to the previous block
+		setBlockState(blocks, world, pos, log.with(getStateFromDirection(direction.getOpposite()), true), boundingBox);
+	}
+
+	protected void tryPlaceLeaves(Set<BlockPos> blocks, ModifiableTestableWorld world, BlockPos.Mutable pos, MutableIntBoundingBox boundingBox) {
+		if (world.testBlockState(pos, BlockState::isAir)) {
+			setBlockState(blocks, world, pos, this.leaves, boundingBox);
+		} else {
+			if (world.testBlockState(pos, isLog -> isLog.getBlock() instanceof SmallLogBlock)) {
+				setBlockState(blocks, world, pos, this.getLog()
+					.with(BareSmallLogBlock.NORTH, world.testBlockState(pos, test -> test.get(BareSmallLogBlock.NORTH)))
```

Comment:
would it be possible to deduplicate the code between here and `getOriginalState`?

Reply:
I don't remember why that didnt work before, but I can give it a go

Reply:
did it work?